lib/pbench/agent/tool_meister.py

```diff
@@ -474,26 +470,28 @@ def _get_data(self):
                 json_str = payload["data"].decode("utf-8")
                 data = json.loads(json_str)
             except Exception:
-                self.logger.warning("data payload in message not JSON, %r", json_str)
+                msg = f"data payload in message not JSON, {json_str!r}"
+                self.logger.warning(msg)
+                self._send_client_status(msg)
```

Comment:
I know this is only two lines; but it's repeated a lot, and refactoring into a "log and reply" or something would help to ensure that we don't accidentally miss some places ... and the future changes in the protocol only need to be made one place. (It'd also reduce your recurring three lines to one, since `msg` would become just the argument to the call.)